iModelCore/Units/src/Formatting/Format.cpp

```diff
@@ -281,6 +281,15 @@ Utf8String Format::FormatQuantity(BEU::QuantityCR qty, BEU::UnitCP useUnit, Utf8
         additionalFormatting = true;
         }
 
+    if (fmtP->GetPresentationType() == PresentationType::FractionalRatio || fmtP->GetPresentationType() == PresentationType::IntegerRatio)
+        {
+        double dval = temp.GetMagnitude();
+        if (fmtP->GetPresentationType() == PresentationType::FractionalRatio)
+            return fmtP->FormatToFractionalRatio(dval);
+        else
+            return fmtP->FormatToIntegerRatio(dval);
+        }
+

```

Comment:
```suggestion
    if (fmtP->GetPresentationType() == PresentationType::FractionalRatio)
        return fmtP->FormatToFractionalRatio(temp.GetMagnitude());
    else if (fmtP->GetPresentationType() == PresentationType::IntegerRatio)
        return fmtP->FormatToIntegerRatio(temp.GetMagnitude());
```
This is shorter. Also I think giving the Format to that method may be a good idea because you may want to respect some of the format's properties (minWidth, ShowLabels and such)